src/hotspot/cpu/x86/stubGenerator_x86_64_arraycopy.cpp

```diff
@@ -2476,6 +2481,281 @@ address StubGenerator::generate_unsafe_copy(const char *name,
 }
 
 
+//  Generate 'unsafe' set memory stub
+//  Though just as safe as the other stubs, it takes an unscaled
+//  size_t argument instead of an element count.
+//
+//  Input:
+//    c_rarg0   - destination array address
+//    c_rarg1   - byte count (size_t)
+//    c_rarg2   - byte value
+//
+// Examines the alignment of the operands and dispatches
+// to an int, short, or byte fill loop.
+//
+address StubGenerator::generate_unsafe_setmemory(const char *name,
+                                                 address byte_fill_entry) {
+  __ align(CodeEntryAlignment);
+  StubCodeMark mark(this, "StubRoutines", name);
+  address start = __ pc();
+
+  // bump this on entry, not on exit:
+  INC_COUNTER_NP(SharedRuntime::_unsafe_set_memory_ctr, rscratch1);
+
+  {
+    Label L_exit, L_fillQuadwords, L_fillDwords, L_fillBytes;
+
+    setup_arg_regs(3);
+#undef dest
+#define dest rdi
+#undef size
+#define size rsi
+#undef wide_value
+#define wide_value rax
+#undef rScratch1
+#define rScratch1 rcx
+#undef byteVal
+#define byteVal rdx
+#undef rScratch3
+#define rScratch3 r8
+#undef rScratch4
+#define rScratch4 r11
```

Comment:
We could do this setup using const Register declaration instead of using #undef/#define pair.

Reply:
We discussed this and the #define option was your preferred method since the registers are being re-used.  Do you want this changed back again?